tools/FhirSchemaManager/Commands/ApplyCommand.cs

```diff
@@ -4,16 +4,97 @@
+using FhirSchemaManager.Exceptions;
+using FhirSchemaManager.Model;
+using FhirSchemaManager.Utils;
 
 namespace FhirSchemaManager.Commands
 {
     public static class ApplyCommand
     {
-        public static void Handler(string connectionString, Uri fhirServer, int version)
+        public static async Task HandlerAsync(string connectionString, Uri fhirServer, int version)
         {
-            Console.WriteLine($"--connection-string {connectionString}");
-            Console.WriteLine($"--fhir-server {fhirServer}");
-            Console.WriteLine($"--version {version}");
+            ISchemaClient schemaClient = new SchemaClient(fhirServer);
+
+            try
+            {
+                List<AvailableVersion> availableVersions = await schemaClient.GetAvailability();
+
+                if (availableVersions?.Any() != true || availableVersions.Count == 1)
+                {
+                    CommandUtils.PrintError(Resources.AvailableVersionsDefaultErrorMessage);
+                    return;
+                }
+                else
+                {
+                    // Removing the current version
+                    availableVersions.RemoveAt(0);
+                }
+
+                availableVersions = availableVersions.Where(availableVersion => availableVersion.Id <= version)
+                    .ToList();
+
+                foreach (AvailableVersion availableVersion in availableVersions)
+                {
+                    string script = await schemaClient.GetScript(availableVersion.Script);
+
+                    await ValidateVersion(schemaClient, availableVersion.Id);
+
+                    // check if the record for given version exists in failed status
+                    SchemaDataStore.ExecuteDelete(connectionString, availableVersion.Id, SchemaDataStore.Failed);
+
+                    // Execute script
+                    SchemaDataStore.ExecuteQuery(connectionString, script, availableVersion.Id);
+
+                    SchemaDataStore.ExecuteUpsert(connectionString, availableVersion.Id, SchemaDataStore.Complete);
+
+                    Console.WriteLine(string.Format(Resources.SuccessMessage, availableVersion.Id));
+                }
+            }
+            catch (SchemaManagerException ex)
+            {
+                CommandUtils.PrintError(ex.Message);
+                return;
+            }
+            catch (HttpRequestException)
+            {
+                CommandUtils.PrintError(string.Format(Resources.RequestFailedMessage, fhirServer));
+                return;
+            }
+            catch (SqlException ex)
+            {
+                CommandUtils.PrintError(string.Format(Resources.QueryExecutionErrorMessage, ex.Message));
+                return;
+            }
+        }
+
+        private static async Task ValidateVersion(ISchemaClient schemaClient, int version)
+        {
+            // to ensure server side polling is completed
+            await Task.Delay(60000);
```

Comment:
Adding a minute delay because server side polling is scheduled at each minute interval. So to ensure it does not read any stale value between schema migration.